src/array.cr

```diff
@@ -1381,6 +1381,18 @@ class Array(T)
     self
   end
 
+  # Replaces the contents of `self` with the contents of *other*.
+  # This resizes the Array to a greater capacity but does not free memory if the given array is smaller.
+  #
+  # ```
+  # a1 = [1, 2, 3]
+  # a1.replace([1])
+  # a1                    # => [1]
+  # a1.remaining_capacity # => 3
+  # a1                    # => [1]
+  # a2.replace([1, 2, 3])
```

Comment:
What is `a2`? The overall code sample would fail not knowing what this variable is

Reply:
solve https://github.com/crystal-lang/crystal/pull/11682/commits/703f6f4930e1d48671f099da81a4a953fbb26515